### ISSUE TITLE
BO: Multistore | Product quantity for all shops

### ISSUE THREAD

**Issue by khouloudbelguith** [PrestaShop/PrestaShop#10274]
**Describe the bug**
http://forge.prestashop.com/browse/BOOM-3495 is not fixed in the develop branch.

**To Reproduce**

1. Enable Multishop
2. Create two shops Shop A and shop Shop B that share quantities in a group 
3. All shops selected => create a standard product & edit the quantity => save => Error

**Screenshots**
![image](https://user-images.githubusercontent.com/16067358/45036642-5176fe00-b05d-11e8-891c-11952d47ee82.png)
**Additionnal information**
PrestaShop version: develop
PHP version: 7.1

**Comment by marionf** [comment]
PR #9145 don't solve this issue

**Comment by sickboards** [comment]
The sharing of stock does not seem relevant in my case (happens with both settings), what I do see if I change the product "customizable mug" to have combinations: 
"main.bundle.js:292 POST http://.../admin/index.php/attribute/generator?_token=0kZM2Oyxy9Wast-d6V2qo2-y8I7hYOdnwcaMpYKCxZw 500 (Internal Server Error) and when updating an existing product:
"VM4311:1 Uncaught SyntaxError: Unexpected token < in JSON at position 0
    at Function.parse [as parseJSON] (<anonymous>)
    at Object.error (form.js?1.7.0:907)
    at u (main.bundle.js:292)
    at Object.fireWith [as rejectWith] (main.bundle.js:292)
    at i (main.bundle.js:292)
    at XMLHttpRequest.<anonymous> (main.bundle.js:292)"

**Comment by nnnext** [comment]
when i switch to a single shop i can update the qty without an error.

**Comment by hugolin69** [comment]
Hi, on a 1.7.6.7, I reach something close.
On a fresh shop, I move to multistore.
I share the quantities, not the customers and orders.
When I try to save the product, after changing the quantity in stock, I have a 500 error, without log, without information. If I re-push "Save", it save it. Always need to save twice.
If I go on a context monostore, it works well.
An y idea?

**Comment by Nukium** [comment]
Hi, we have the same issue on 1.7.6.9.

Error when saving a product after changing the quantity in context "all shops" but the quantity is still modified.

Thank you

**Comment by SNor42** [comment]
Hi!
To be true I don't know what `$stockMvtRepository = $sfContainer->get('prestashop.core.api.stock_movement.repository'); ` do, but in my case the problem occurred when this code was called. So what I made was:
Under _src/Core/Stock/StockManager.php_ replaced this:
```
if ($stockMvt) {
    $sfContainer = SymfonyContainer::getInstance();
    if (null !== $sfContainer) {
      $stockMvtRepository = $sfContainer->get('prestashop.core.api.stock_movement.repository');
      return $stockMvtRepository->saveStockMvt($stockMvt);                   
    }
}
```
into this:
```
if ($stockMvt) {
    $sfContainer = SymfonyContainer::getInstance();
    if (null !== $sfContainer) {
      try {
          $stockMvtRepository = $sfContainer->get('prestashop.core.api.stock_movement.repository');
          return $stockMvtRepository->saveStockMvt($stockMvt);
      } catch (\Exception $e) {
          error_log($e->getMessage());
          return false;
      }                    
    }
}
```
Could somebody check this solution? I really not sure that this fix won't do something bad somewhere else!

**Comment by jf-viguier** [comment]
> Hi! To be true I don't know what `$stockMvtRepository = $sfContainer->get('prestashop.core.api.stock_movement.repository'); ` do, but in my case the problem occurred when this code was called. So what I made was: Under _src/Core/Stock/StockManager.php_ replaced this:
> 
> ```
> if ($stockMvt) {
>     $sfContainer = SymfonyContainer::getInstance();
>     if (null !== $sfContainer) {
>       $stockMvtRepository = $sfContainer->get('prestashop.core.api.stock_movement.repository');
>       return $stockMvtRepository->saveStockMvt($stockMvt);                   
>     }
> }
> ```
> 
> into this:
> 
> ```
> if ($stockMvt) {
>     $sfContainer = SymfonyContainer::getInstance();
>     if (null !== $sfContainer) {
>       try {
>           $stockMvtRepository = $sfContainer->get('prestashop.core.api.stock_movement.repository');
>           return $stockMvtRepository->saveStockMvt($stockMvt);
>       } catch (\Exception $e) {
>           error_log($e->getMessage());
>           return false;
>       }                    
>     }
> }
> ```
> 
> Could somebody check this solution? I really not sure that this fix won't do something bad somewhere else!

Try to post a Pull Request and a maintainer will answer

**Comment by SNor42** [comment]
Update!
I've checked these line:
$sfContainer->get('prestashop.core.api.stock_movement.repository');
It calls the: src\PrestaShopBundle\Entity\Repository\StockManagementRepository.php
So in the __construct class there these a line:
```
if ($shop->getContextType() !== $shop::CONTEXT_SHOP) {
throw new NotImplementedException('Shop context types other than "single shop" are not supported');
}
$this->shopId = $shop->getContextualShopId();
```
That's why it doesn't work!
So I've changed to:
```
if ($shop->getContextType() !== $shop::CONTEXT_SHOP) {
// throw new NotImplementedException('Shop context types other than "single shop" are not supported');
$this->shopId = 1;
} else {
$this->shopId = $shop->getContextualShopId();
}
```
... and works. And also it logs the stock movement when we are in all shops. Finally!

**Comment by codeway** [comment]
Same issue with 1.7.8.7.
Multistores (x4).
When i tried to update a product having multiple attributes combinaisons.
The error is triggered "Shop context types other than "single shop" are not supported".
The first attrib is updated (like stock quantity), but not the following.
With your fix @SNor42 it's seems to work, but this fix is in the Presta Core.
Is prestashop team had fixed this bug in the Core or do we have another way to override this behavior ?

**Comment by SNor42** [comment]
Hi, Before the fix, it only saved the combinations for me one by one (needed to click on multiple times on save).
I didn't made a PR, because, I'm not sure this is the correct way of fixing.

**Comment by medjabir** [comment]
I am also facing the same bug in `1.7.8.8`. I managed to implement a fix which required an edit of the core file `StockManager`.

**Comment by MatShir** [comment]
Seems to be fixed in 8.1 :)

**Comment by djoelleuch** [comment]
Hello @MatShir ,
The issue is fixed on 8.1 as you can see below :

[issu1072.webm](https://github.com/PrestaShop/PrestaShop/assets/115152460/9f7a7f73-1e9e-48b5-ac4b-fe714dbbb839)

I will close the issue , Thank you .
